apps/src/lib/wasm_loader/mod.rs

```diff
@@ -260,67 +261,36 @@ pub async fn pre_fetch_wasm(wasm_directory: impl AsRef<Path>) {
 pub fn read_wasm(
```

Comment:
maybe this could be `try_read_wasm` and we could provide another `read_wasm` from it that handles the error and exits

Reply:
I'd be more inclined to use "read_wasm" for this here returning a `Result` then "read_wasm_or_exit" for the one which exits, what do you think?

At the moment we have many places in the code where the process can terminate, most of the places where this happens I don't think should be the concern of that component e.g. should the `wasm_loader` module ever be deciding to terminate the process or not?  So eventually we could have most calls returning `Result<T>` rather than returning `T` or maybe exiting altogether, in that case we could want to make explicit where exiting might happen with a suffix/prefix in the name rather than the other way round, if we followed an idiom like: returning a `Result<T>` means this function won't suddenly exit. Returning a `T` it shouldn't suddenly exit either unless it has a string in the name like `or_exit` (I'm not sure how to encode this in the type signature)

Reply:
I've added `read_wasm_or_exit` and used it in a couple of places where we were previously panicking or exiting. Ultimately this should become something like `read_wasm_or_panic` I think (or go away altogether) but that can happen in another PR at some point